### PR TITLE
[Refactoring] Jmap Mailbox contracts cleanup

### PR DESCRIPTION
**Pull request by Arsnael**

### CODE REVIEW

File: server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala [hunk 1]
```diff
@@ -58,6 +58,7 @@ trait MailboxSetMethodContract {
   }
 
   def randomMailboxId: MailboxId
+  def errorInvalidMailboxIdMessage: String
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621-integration-tests/jmap-rfc-8621-integration-tests-common/src/main/scala/org/apache/james/jmap/rfc8621/contract/MailboxSetMethodContract.scala:61]: Not sure if this is ok or if overriding the test method in `DistributedMailboxSetMethodTest` would be better?

**Comment by chibenwa** [comment]
(can we rebase & apply this cleanup to the recent merges?)

**Comment by chibenwa** [comment]
`docker: Error response from daemon: No such container: git-publish-jmap-contract-clean-15a16d53-f967-413a-a514-a70fd044f61c.`

mvn tests are :green_apple: 

test this please

**Review by LanKhuat — APPROVED** [review]

**Review by nvduc91 — APPROVED** [review]

**Review by chibenwa — APPROVED** [review]

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**